src/goFormat.ts

```diff
@@ -23,6 +24,13 @@ export class GoDocumentFormattingEditProvider implements vscode.DocumentFormatti
 		let formatTool = goConfig['formatTool'] || 'goreturns';
 		let formatFlags = goConfig['formatFlags'].slice() || [];
 
+		isModSupported(document.uri).then(isMod => {
+			formatTool = 'goimports';
```

Comment:
`isModSupported` returns a promise. Promises continue with the `then` function asynchronously. Which means that the `formatTool` variable will be updated _after_ the `runFormatter` (see a few lines below) function has been called. So it probably does not what you wanted it to do :)

You have two options now: 
* move all the code after the `catch` (so starting with the `// We ignore ...` up to the `return this.runFormatter(...`) into the `then` function here
* or use [async/await](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await) to wait for the promise to be done and continues with the rest of the code